.github/copilot-instructions.md

```diff
@@ -0,0 +1,238 @@
+# IDE Sidecar - AI Coding Agent Instructions
+
+## Project Overview
+
+IDE Sidecar is a **Java 21 Quarkus application** that serves as the backend for Confluent's VS Code extension. It exposes REST and GraphQL APIs for managing connections to Kafka clusters and consuming/producing messages. The project is compiled to a **native executable using GraalVM** for fast startup and low memory footprint.
+
+**Key Connection Types:**
+- `CCLOUD`: Confluent Cloud connections (OAuth-based authentication)
+- `LOCAL`: Confluent Local/standalone Kafka
+- `DIRECT`: Direct connections to Kafka clusters (supports multiple auth methods: Basic, SASL/SCRAM, Kerberos, OAuth, TLS)
+
+## Architecture
+
+### Processor Chain Pattern
+
+Request handling uses a **chain-of-responsibility pattern** via the `Processor<T, U>` class. Processors are chained using `Processor.chain(...)`:
+
+```java
+// Example from MessageViewerProcessorBeanProducers
+Processor.chain(
+    new ConnectionProcessor<>(connectionManager),
+    new KafkaClusterInfoProcessor<>(clusterCache),
+    new ConsumeStrategyProcessor(nativeStrategy, ccloudStrategy),
+    new EmptyProcessor<>()
+)
+```
+
+Each processor:
+1. Receives a context object (e.g., `KafkaRestProxyContext`, `ProxyContext`)
+2. Performs its operation (auth, cluster lookup, proxying)
+3. Passes the modified context to the next processor
+
+**Find processor implementations in:** `src/main/java/io/confluent/idesidecar/restapi/processors/` and `src/main/java/io/confluent/idesidecar/restapi/proxy/`
+
+### GraphQL API Structure
+
+GraphQL APIs are defined with `@GraphQLApi` annotation:
+- `ConfluentCloudQueryResource`: CCloud organizations, environments, Kafka clusters, Schema Registries
+- `ConfluentLocalQueryResource`: Local Kafka and Schema Registry
+- `DirectQueryResource`: Direct connection clusters
+
+Schema generation happens automatically. Find generated schemas in `src/generated/resources/`.
+
+### REST API Structure
+
+REST endpoints use JAX-RS annotations (`@Path`, `@POST`, etc.):
+- `ConnectionsResource`: CRUD for connections (`/gateway/v1/connections`)
+- `KafkaConsumeResource`: Message viewer (`/gateway/v1/clusters/{cluster_id}/topics/{topic_name}/partitions/-/consume`)
+- `ClusterRestProxyResource`: Proxies requests to Kafka/SR clusters
```

Comment:
Should we add other endpoints, like the Preferences API (`/gateway/v1/preferences`), too?